### ISSUE TITLE
Variables bound in for loops are being type checked incorrectly

### ISSUE THREAD

**Issue by leafpetersen** [dart-lang/dev_compiler#20]
This code should produce an error (even in regular Dart), but seems to be currently passing our typechecker.

```
  for (int i = 0; i < 10; i++) {
    i = /*pass should be severe:StaticTypeError*/"hi";
  }
```
